control/timeresp.py

```diff
@@ -826,60 +830,187 @@ def impulse_response(sys, T=None, X0=0., input=0, output=None, T_num=None,
     return T, yout
 
 # utility function to find time period and time increment using pole locations
-def _ideal_tfinal_and_dt(sys):
-    constant = 7.0
-    tolerance = 1e-10
-    A = ssdata(sys)[0]
-    if A.shape == (0,0): 
-        # no dynamics
-        tfinal = constant * 1.0 
-        dt = sys.dt if isdtime(sys, strict=True) else 1.0
-    else:
-        poles = sp.linalg.eigvals(A)
-        if isdtime(sys, strict=True):
-            poles = np.log(poles)/sys.dt # z-poles to s-plane using s=(lnz)/dt
-        
-        # calculate ideal dt
-        if isdtime(sys, strict=True):
-            dt = sys.dt
-        else:
-            fastest_natural_frequency = max(abs(poles))
-            dt = 1/constant / fastest_natural_frequency
+def _ideal_tfinal_and_dt(sys, is_step=True):
+    """helper function to compute ideal simulation duration tfinal and dt, the 
+    time increment. Usually called by _default_time_vector, whose job it is to 
+    choose a realistic time vector. Considers both poles and zeros.  
+    
+    For discrete-time models, dt is inherent and only tfinal is computed.
+
+    Parameters
+    ----------
+    sys : StateSpace or TransferFunction
+        The system whose time response is to be computed
+    is_step : bool
+        Scales the dc value by the magnitude of the nonzero mode since
+        integrating the impulse response gives ∫exp(-λt) = -exp(-λt)/λ.
```

Comment:
Python 2 encoding error

```suggestion
        integrating the impulse response gives :math:`\int e^{-\lambda t} = -e^{-\lambda t}/ \lambda`
```